src/ComponentProvider.ts

```diff
@@ -0,0 +1,90 @@
+import type { Schema, Message, AsyncAPIDocument, ChannelParameter } from '@asyncapi/parser';
+
+/**
+ * This Singleton class will provide all sorts of data for optimizers.
+ *
+ * @public
+ */
+export class ComponentProvider {
+  messages = new Map<string, Message>();
+  schemas = new Map<string, Schema>();
+  parameters = new Map<string, ChannelParameter>();
+
+  private static instance: ComponentProvider;
+  private constructor(private document: AsyncAPIDocument) {
+    this.scanChannels();
+    this.scanComponents();
+  }
+  static getInstance(document: AsyncAPIDocument): ComponentProvider {
+    if (!ComponentProvider.instance) {
+      ComponentProvider.instance = new ComponentProvider(document);
+    }
+    return ComponentProvider.instance;
+  }
+
+  private scanChannels = (): void => {
```

Comment:
@KhudaDad414 Lets try to format the code in a more readable fashion. My personal favorites which make a ton of difference are:
1. Blank line before `return` unless it is the first line of the block.
2. Blank line before and after `if` unless it is the first line of the block.
3. Blank line before & after a set of `const`  unless it is the first line of the block.
4. Whitespace after `,` when passing parameters or defining arrays

Reply:
We can extend the `eslint` rules to achieve that. I don't think so that it should be done manually. Everyone has another manner to write the code and we should use tool to achieve the consistency in the remote source. @KhudaDad414 Could you create issue for that?